Updates to Rereco2016

DouglasBurns opened the pull request:
ElectronID has been updated.
PseudoTop dependency has been removed
Electron channel uses new Lumi json files + updates in their crab configs.
BTagSF now use ICHEP recommendations
Updated Muon Triggers
Updated a MET filter (Seems we may have more to add)
Add a git cms-merge-topic for running MET corrections 


This PR includes a lot of back and forth in the electron trigger. (i.e. between Ele32 unprescale and Ele27 partial prescale)
This includes updates in Globaltags, lumi jsons, triggers etc and so would be uncompatible with the ICHEP dataset. Perhaps this should for now be in a new branch of NTupleProduction?

kreczko commented on data/JSON/LSfor_HLT_Ele27_WPTight_Gsf_withLowestSeed_L1_SingleIsoEG26.json:
```diff
@@ -0,0 +1,377 @@
+
```
This is not a proper JSON file

kreczko commented on python/Jets_Setup_cff.py:
```diff
@@ -39,7 +39,7 @@ def setup_jets(process, cms, options, postfix="PFlow"):
         from CondCore.DBCommon.CondDBSetup_cfi import *
         import os
-        era = "Fall15_25nsV2_"
+        era = "Spring16_25nsV6_"
```
A new era begins

kreczko commented on python/ntp/commands/run/analysis/DICE.py:
```diff
@@ -208,4 +227,4 @@ def create_job_layer(self, input_files, mode):
             jobs.append(job)
 
-        return jobs
+        return jobs
```
New lines at the end of a file are also useful

kreczko commented on python/ntp/commands/run/analysis/__init__.py:
```diff
@@ -57,6 +57,24 @@
 ]
 
 
+def input_files_from_path(path):
```
this should be available in `ntp.commands.run`. Why duplicate it?

kreczko commented on python/ntp/commands/run/analysis/template.py:
```diff
@@ -163,7 +169,7 @@ def getAnalysisSettings(analysisMode):
 
 # integrated luminosity the MC simulation will be scaled to
-lumi = 15933
+lumi = 12878
```
we have now less lumi?

kreczko commented on python/userdata/JetUserData_cfi.py:
```diff
@@ -6,7 +6,7 @@
     beamSpotCollection=cms.InputTag('offlineBeamSpot'),
     # Top Object Definitions
-    minSignalJetPt=cms.double(20.),
+    minSignalJetPt=cms.double(30.),
```
Just need to keep this in mind. Affects HT distribution.

DouglasBurns commented:
Removed JSON Electron JSON that wasnt JSON (dont need anyway if we are using Ele32)
Removed duplicated function
Addes Moriond golden JSON
Added a couple more DY datasets from current tranche (though old tune)
Update Lumi
